internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -496,6 +509,32 @@ func (r *BMACReconciler) reconcileAgentInventory(log logrus.FieldLogger, bmh *bm
 
 }
 
+// Ask BMH to reboot the host if the agent is unbound after installation
+//
+// By re-attaching the BMH and clearing the Image field on it, BMAC will clear
+// the Image data to force the boot from ISO
+//
+func (r *BMACReconciler) reconcileUnboundAgent(log logrus.FieldLogger, bmh *bmh_v1alpha1.BareMetalHost, agent *aiv1beta1.Agent) reconcileResult {
+	log.Debugf("Started Unbound Agent reconcile for agent %s/%s and bmh %s/%s", agent.Namespace, agent.Name, bmh.Namespace, bmh.Name)
+
+	// proceed with the reconcile only when the agent ask for user action following unbinding
+	boundCondition := conditionsv1.FindStatusCondition(agent.Status.Conditions, aiv1beta1.BoundCondition)
+	if boundCondition == nil || boundCondition.Reason != aiv1beta1.UnbindingPendingUserActionReason {
+		return reconcileComplete{}
+	}
+
+	// re-attach the bmh and clear the ISO url to force ironic image refresh
+	// and re-conciliation of BMH and agent. Also, clear the hw details just
+	// in case. They will be regenerated in the reconciles following the agent's
+	// reboot
+	delete(bmh.ObjectMeta.Annotations, BMH_DETACHED_ANNOTATION)
+	delete(bmh.ObjectMeta.Annotations, BMH_HARDWARE_DETAILS_ANNOTATION)
+	bmh.Spec.Image = nil
+
+	log.Debugf("Unbound Agent reconciled to BMH \n %v \n %v", agent, bmh)
```

Comment:
I think that we can have this in Info. It is important enough and not going to happen a lot of times.